dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/programrule/implementers/event/ShowErrorExecutor.java

```diff
@@ -25,28 +25,28 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.hisp.dhis.tracker.programrule.implementers;
+package org.hisp.dhis.tracker.programrule.implementers.event;
 
 import static org.hisp.dhis.tracker.programrule.IssueType.ERROR;
 
-import org.hisp.dhis.rules.models.RuleActionShowError;
+import java.util.Optional;
+
+import lombok.RequiredArgsConstructor;
+
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Event;
 import org.hisp.dhis.tracker.programrule.IssueType;
-import org.springframework.stereotype.Component;
+import org.hisp.dhis.tracker.programrule.ProgramRuleIssue;
 
 /**
- * This implementer show errors calculated by Rule Engine.
+ * This executor shows errors calculated by Rule Engine.
```

Comment:
Just for me to understand rule engine better 😄 How does it show an error? Is it that it will add an error to the validation report returned to the user? And if so in what case?

Reply:
This show error is an executor, it means that rule engine execution computed it as an action to apply to the given enrollment/event.
This executor is simply going to add an error to the report returned to the user.
This particular executor has no conditions, it will add the error. Other executors like `ShowErrorOnCompleteExecutor` will add the error only if the enrollment/event status is `COMPLETED`